tls/s2n_config.c

```diff
@@ -232,15 +232,24 @@ int s2n_config_defaults_init(void)
     if (s2n_is_in_fips_mode()) {
         POSIX_GUARD(s2n_config_init(&s2n_default_fips_config));
         POSIX_GUARD(s2n_config_setup_fips(&s2n_default_fips_config));
+        if (!s2n_default_fips_config.loaded_system_certs) {
+            POSIX_GUARD(s2n_config_load_system_certs(&s2n_default_fips_config));
```

Comment:
There were some s2n_init tests failing, and I thought s2n_init was being called more than once, but really cleaning up the default certs did not properly reset the loaded_system_certs flag. I moved the flag closer to the trust store so clearing the default certs also resets this flag, and this is tested now in the s2n_init tests that initialize and free the default configs.

I remember there being customers that needed to call s2n_init more than once (I think because a dependency also called s2n_init). This change will mean that a double s2n_init will now fail here. Is this a concern?